### PR TITLE
get user and group information using securejoin and runc's user library

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/container_internal.go [hunk 1]
```diff
@@ -1065,7 +1063,7 @@ func (c *Container) generatePasswd() (string, error) {
 		return "", errors.Wrapf(err, "unable to read passwd file %s", originPasswdFile)
 	}
 
-	pwd := fmt.Sprintf("%s%d:x:%d:%d:container user:%s:/bin/sh\n", orig, uid, uid, gid, c.WorkingDir())
+	pwd := fmt.Sprintf("%s%d:x:%d:%d:container user:%s:/bin/sh\n", orig, uid, uid, execUser.Gid, c.WorkingDir())
```
Comment (TomSweeneyRedHat) [inline libpod/container_internal.go]: Should we be using execUser.Uid here instead of uid?

File: pkg/lookup/lookup.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	for _, gid := range gids {
+		uintgids = append(uintgids, uint32(gid))
+	}
+	return uintgids, nil
+}
+
+// User takes a containermount path and user name or id and returns
+// a matching User structure from /etc/passwd.  If it cannot locate a user
+// with the provided information, an ErrNoPasswdEntries is returned.
+func User(containerMount, userIDorName string) (*user.User, error) {
```
Comment (vrothberg) [inline pkg/lookup/lookup.go]: Can we prefix this with `Get` to be consistent with the upper functions?

File: pkg/lookup/lookup.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	if len(users) > 0 {
+		return &users[0], nil
+	}
+	return nil, user.ErrNoPasswdEntries
+}
+
+// Group takes ac ontainermount path and a group name or id and returns
+// a match Group struct from /etc/group.  if it cannot locate a group,
+// an ErrNoGroupEntries error is returned.
+func Group(containerMount, groupIDorName string) (*user.Group, error) {
```
Comment (vrothberg) [inline pkg/lookup/lookup.go]: Same here: Can we prefix this with `Get` to be consistent with the upper functions?

**Comment by vrothberg** [comment]
Code LGTM (besides the minor nitpicks) but the commit needs a proper description. I know it's very common on Github to not describe commits but I find it very important for reviewing, studying and also for maintenance.

File: pkg/lookup/lookup.go (hunk 2 above)
Comment (baude) [reply pkg/lookup/lookup.go]: fixed

File: pkg/lookup/lookup.go (hunk 3 above)
Comment (baude) [reply pkg/lookup/lookup.go]: fixed

**Comment by baude** [comment]
addressed @vrothberg comments

**Comment by mheon** [comment]
LGTM

**Comment by mheon** [comment]
bot, retest this please

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, some commit verbiage would be nice.  Tests don't look to be hip at the moment though.

**Pull request merged**